deploy/06_asset/09_deploy_asset.ts

```diff
@@ -1,21 +1,102 @@
 import {HardhatRuntimeEnvironment} from 'hardhat/types';
-import {DeployFunction} from 'hardhat-deploy/types';
-
-const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
-  const {deployments, getNamedAccounts} = hre;
-  const {deploy} = deployments;
+import fs from 'fs-extra';
+import {DeployFunction, DeploymentSubmission} from 'hardhat-deploy/types';
+import {BigNumber} from '@ethersproject/bignumber';
+import {getAddress} from '@ethersproject/address';
 
+const func: DeployFunction = async function (
+  hre: HardhatRuntimeEnvironment
+): Promise<void> {
+  const {deployments, getNamedAccounts, getChainId, upgrades, ethers} = hre;
   const {deployer} = await getNamedAccounts();
+  const {log} = deployments;
+
+  const chainId = await getChainId();
 
   const sandContract = await deployments.get('Sand');
 
-  await deploy('Asset', {
-    from: deployer,
-    args: [sandContract.address, deployer, deployer],
-    log: true,
-    skipIfAlreadyDeployed: true,
-  });
+  const assetContract = await deployments.getOrNull('Asset');
+  if (assetContract) {
+    log(`Asset already deployed at ${assetContract.address}`);
+    return;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const Asset = await (ethers as any).getContractFactory('Asset', deployer); // TODO check types with hardhat-ethers and hardhat-deploy-ethers, for now use `any`
+
+  // Problem : cannot use deterministic deployment with openzepelin proxies ?
```

Comment:
Is this still an issue?

Reply:
yes, hardhat-upgrades do not allow for it yet. Some discussion here about integrating hardhat-upgrades and hardhat-deploy: https://forum.openzeppelin.com/t/deterministic-deployment-using-upgrades-plugin/5469